pkg/vsphere/tasks/waiter.go

```diff
@@ -41,21 +46,21 @@ type Task interface {
 
 // Wait wraps govmomi operations and wait the operation to complete
 // Sample usage:
-//    info, err := Wait(ctx, func(ctx) (*TaskInfo, error) {
-//       return vm.Reconfigure(ctx, config)
+//    info, err := Wait(ctx, func(ctx), sess *session.Session, *object.Reference, (*object.Reference, *TaskInfo, error) {
+//       return vm, vm.Reconfigure(ctx, sess, target, config)
 //    })
-func Wait(ctx context.Context, f func(context.Context) (Task, error)) error {
-	_, err := WaitForResult(ctx, f)
+func Wait(ctx context.Context, sess *session.Session, target object.Reference, f func(context.Context) (Task, error)) error {
+	_, err := WaitForResult(ctx, sess, target, f)
 	return err
 }
 
 // WaitForResult wraps govmomi operations and wait the operation to complete.
 // Return the operation result
 // Sample usage:
-//    info, err := WaitForResult(ctx, func(ctx) (*TaskInfo, error) {
-//       return vm.Reconfigure(ctx, config)
+//    info, err := WaitForResult(ctx, sess *session.Session, *object.Reference, func(ctx) (*TaskInfo, error) {
+//       return vm, vm.Reconfigure(ctx, config)
 //    })
-func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
+func WaitForResult(ctx context.Context, sess *session.Session, target object.Reference, f func(context.Context) (Task, error)) (*types.TaskInfo, error) {
```

Comment:
`TaskInfo.Entity` property holds the "target" of the task, so you can get rid of that parameter as well. See https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.TaskInfo.html and https://sourcegraph.com/github.com/vmware/vic@master/-/blob/vendor/github.com/vmware/govmomi/vim25/types/types.go#L39888